recipe/meta.yaml

```diff
@@ -132,7 +133,7 @@ outputs:
         - zstd
       run_constrained:
         - apache-arrow-proc =*={{ build_ext }}
-        - cudatoolkit >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
+        - __cuda >= 9   # [cuda_compiler_version != "None"]
```

Comment:
I cannot suggest across a changed line, but could you respond about the [point](https://github.com/conda-forge/arrow-cpp-feedstock/pull/1079#discussion_r1222321652) of moving this to `run:`?

Reply:
It wasn't clear if we wanted to make it hard requirement as opposed to a soft one. So stuck with the status quo

As conda & mamba (relatively) recently learned about `__cuda` and some users may still be on old version of these tools, there can be some value in going with the soft requirement (so they can still install). That said, maybe they can workaround these constraints. Some related discussion ( https://github.com/conda-forge/conda-forge.github.io/issues/1840#issuecomment-1318159269 )

Reply:
`__cuda` is a run-constraint in a number of packages already, notably tensorflow. The discussion on this was very spread out. Here's the original [issue](https://github.com/conda-forge/tensorflow-feedstock/issues/174), more discussion happened [here](https://github.com/conda-forge/conda-forge.github.io/pull/1601)

Reply:
Yep don't have strong feelings. Just pointing out the tradeoffs. If we are ok with those tradeoffs, great 👍

Reply:
OK, thanks. Given that Keith and I have been in agreement about this approach (which has also been gestating for quite a while in https://github.com/conda-forge/arrow-cpp-feedstock/issues/942 already), I think it's fine to leave this as it is now.